padertorch/train/hooks.py

```diff
@@ -816,3 +817,43 @@ def close(self, trainer):
         if self.onset_value is not None:
             module = self.get_module(trainer)
             setattr(module, self.name[-1], self.onset_value)
+
+
+class LRSchedulerHook(TriggeredHook):
+    """
+    A hook that applies a learning rate scheduler from `torch.optim.lr_scheduler`
+    to the training.
+
+    Examples:
+        >>> trainer = pt.Trainer(...)   # doctest: +SKIP
+        >>> trainer.register_hook(LRSchedulerHook(
+        ...     torch.optim.lr_scheduler.StepLR(
+        ...         trainer.optimizer.optimizer, step_size=2, gamma=0.98)
+        ... ))  # doctest: +SKIP
+
+    Note:
+
+        This hook can only be used with learning rate schedulers that
+        don't require metrics.
+
+    """
+    # It is very likely that this check is exclusive to this hook
+    IS_PYTORCH_1_1 = LooseVersion(torch.__version__) >= '1.1.0'
+
+    def __init__(self, lr_scheduler, trigger=(1, 'epoch')):
+        super().__init__(trigger)
+        self.lr_scheduler = lr_scheduler
+
+    def pre_step(self, trainer: 'pt.Trainer'):
+        if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch):
+            if trainer.epoch > 0 or not self.IS_PYTORCH_1_1:
```

Comment:
should the step be called if pytorch larger 1.1.0 or if it is not?

Reply:
It should be skipped in the first epoch if pytorch is larger than 1.1.0

Reply:
so it should be an and?

Reply:
ah okay, to prevent the warning. Then it is fine

Reply:
It does not only prevent the warning. The learning rate schedule would be shifted by one epoch otherwise